src/test/java/org/candlepin/model/ProductTest.java

```diff
@@ -406,9 +406,9 @@ public void testSetProvidedProductsChecksForCyclesOnProvidedProducts(int depth)
         Product p3 = spy(new Product())
             .setId("p3");
 
-        doReturn(1).when(p1).getEntityVersion();
```

Comment:
There's no need to mock these. In fact we don't even need to spy on them anymore.

Reply:
I don't understand. We're ensuring the output of getEntityVersion is fixed so that the scenario where we have sets of children that have the same sum don't cause a collision.

I opted for `spy` here instead of `mock` so I didn't have to worry about mocking or breaking anything else that is used in the mapping when it's added as a child of another product.

Reply:
Added some comments around this to clarify the purpose of the mocking of `.getEntityVersion()`